src/elo/util/dbInterface.ts

```diff
@@ -0,0 +1,228 @@
+import { 
+  ComputedElo as ComputedEloIM,
+  EloSnapshot as EloSnapshotIM,
+  Submatch as SubmatchIM,
+  Match as MatchIM
+} from "../util/types";
+
+import { GraphQLContext } from "../../context";
+import { Game, Match, Result, ComputedElo } from "@prisma/client";
+
+
+export const pullComputedElos = async (
+  playerIds: string[],
+  context: GraphQLContext
+): Promise<ComputedEloIM[]> => {
+
+  const e = await context.prisma.computedElo.findMany({
+    where: {
+      playerId: {
+        in: playerIds,
+      },
+      versionId: context.eloVersion.id,
+    },
+  });
+
+  const elos: ComputedEloIM[] = [];
+  const newPlayers: string[] = [];
+  
+  // populate absent computedElos with null
+  playerIds.forEach(id => {
+    let found = false;
+
+    for (let i = 0; i < e.length; ++i) {
+      if (e[i].playerId == id) {
+        elos.push(<ComputedEloIM>(e[i]));
+        found = true;
+        break;
+      }
+    }
+
+    if (!found && newPlayers.indexOf(id) === -1)
+      newPlayers.push(id);
+
+  });
+
+  (await newComputedElos(newPlayers, context)).forEach(n => {
+    elos.push(n);
+  });
+
+  return elos;
+
+};
+
+export const pullMatches = async (
+  oldestMatchId: string,
+  context: GraphQLContext
+): Promise<MatchIM[]> => {
+
+  const match = await context.prisma.match.findUnique({
+    where: {
+      id: oldestMatchId
+    }
+  });
+
+  if (match == null)
+    return [];
+
+  // get all matches newer than that
+  const m = await context.prisma.match.findMany({
+    where: {
+      timestamp: {
+        gte: match.timestamp
+      }
+    },
+    include: {
+      games: {
+        include: {
+          results: true,
+        }
+      }
+    }
+  });
+
+  return m.map(match => matchConverter(match));
+};
+
+// pulls all computed Elo values in range, of min,max (inclusive)
+export const pullComputedElosInRange = async (
+  min: number,
+  max: number,
+  context: GraphQLContext
+): Promise<ComputedElo[]> => {
+  
+  const e = await context.prisma.computedElo.findMany({
+    where: {
+      AND: [
+        { elo: { lte: max } },
+        { elo: { gte: min } },
+      ],
+    },
+  });
+
+  return e;
+};
+
+const newComputedElos = async (
```

Comment:
Thoughts on `createNewComputedElos` to clarify that it's a create operation?

Reply:
I think that's probably smart - it wasn't initially a create operation, that change came later as part of a refactor to fix behavior in the overall pipeline. When I updated the functionality, I didn't change the name. Will do! :D